packages/babel-types/src/clone/cloneNode.js

```diff
@@ -53,14 +58,18 @@ export default function cloneNode<T: Object>(node: T, deep: boolean = true): T {
     for (const field of Object.keys(NODE_FIELDS[type])) {
       if (has(node, field)) {
         newNode[field] = deep
-          ? cloneIfNodeOrArray(node[field], true)
+          ? cloneIfNodeOrArray(node[field], true, withoutLoc)
           : node[field];
       }
     }
   }
 
   if (has(node, "loc")) {
-    newNode.loc = node.loc;
+    if (withoutLoc) {
+      newNode.loc = null;
+    } else {
+      newNode.loc = node.loc;
+    }
   }
   if (has(node, "leadingComments")) {
     newNode.leadingComments = node.leadingComments;
```

Comment:
when both `deep` and `withoutLoc` is `true`, we should apply `cloneIfNodeOrArray` on these comment nodes, same applied to `innerComments` and `trailingComments`.

Reply:
something like this? 

```
newNode.leadingComments = deep && withoutLoc ? cloneIfNodeOrArray(node.leadingComments, true, true) : node.leadingComments;

```

Reply:
Yep! And please add a test on the new behaviour.

Reply:
Done!